### DISCUSSION TITLE
Missing ops?

### DISCUSSION THREAD

**Discussion by pierrec** [maximecb/uvm#33] [Q&A]
From the current set of integer sign extensions and f32 conversion codes, what about the following ones?
sx_i8_i16
i16_to_f32
i8_to_f32

**Comment by maximecb** [comment]
Hi Pierre!

These ops aren't really missing. I tried to keep the number of operations down on purpose. Both to simplify the implementation of the VM, and also to favor operations that map well to current hardware.

`sx_i8_i16` => You can sign-extend with `sx_i8_i32`. Then you can either just ignore the higher bits, or truncate with `trunc_u16` if you need to discard them.

`i16_to_f32`, `i8_to_f32` => For this you can sign-extend (if desired) and then use `i32_to_f32`

Curious to know what you are working on? :)

**Comment by pierrec** [comment]
OK it makes sense. Sorry for the basic questions, this is the first time I dabble into writing a compiler, but it is a LOT of fun!
I am writing a Go compiler that targets UVM :) (well a subset of the Go specs as go routines for instance will not be supported). Making good progress but still a loooong way to go (pun intended).

**Reply by maximecb** [reply]
It's no problem, you can ask as many questions as you want. I'm glad to see people are doing things with UVM (and having fun in the process) :)
